conformance/experimental_conformance_test.go

```diff
@@ -119,32 +117,16 @@ func testExperimentalConformance(t *testing.T) {
 			},
 			Implementation:      *implementation,
 			ConformanceProfiles: conformanceProfiles,
+			ReportOutput:        *flags.ReportOutput,
 		})
 	if err != nil {
 		t.Fatalf("error creating experimental conformance test suite: %v", err)
 	}
 
 	cSuite.Setup(t)
 	cSuite.Run(t, tests.ConformanceTests)
-	report, err := cSuite.Report()
+	_, err = cSuite.Report(t)
 	if err != nil {
 		t.Fatalf("error generating conformance profile report: %v", err)
 	}
-	writeReport(t.Logf, *report, *flags.ReportOutput)
-}
-
-func writeReport(logf func(string, ...any), report confv1a1.ConformanceReport, output string) error {
-	rawReport, err := yaml.Marshal(report)
-	if err != nil {
-		return err
-	}
-
-	if output != "" {
-		if err = os.WriteFile(output, rawReport, 0644); err != nil {
-			return err
-		}
-	}
-	logf("Conformance report:\n%s", string(rawReport))
-
-	return nil
```

Comment:
Why move all of this? You could also achieve this by passing `cSuite.ReportOutput` to the `writeReport` call on line 133, which would keep the output logic in the containing block, keeping the `Report` call more simple and closer to single-purpose.

Reply:
My idea is that, when gwapi implementations want to get the Report and use the experiment test suite, the flag `cSuite.Report` is not used inside the `experiment test suite` itself, we need to rewrite another writeReport to achieve this.